### PR TITLE
Object Detection Support

### PR DESCRIPTION
**Pull request by joyce-yuan**
Implemented the object detection + classification task

This implementation follows the implementation from [Geeks for Geeks](https://www.geeksforgeeks.org/yolov3-from-scratch-using-pytorch/), link here: https://www.geeksforgeeks.org/yolov3-from-scratch-using-pytorch/

To run, download the 2012 Pascal dataset: http://host.robots.ox.ac.uk/pascal/VOC/voc2012/index.html, and specify where the dataset is in `dpath` in `sys_config`.

### CODE REVIEW

File: src/utils/model_utils.py [hunk 1]
```diff
@@ -70,29 +74,91 @@ def train(self,
...
-            loss.backward()
-            optim.step()
-            train_loss += loss.item()
-            pred = output.argmax(dim=1, keepdim=True)
-            # view_as() is used to make sure the shape of pred and target are
-            # the same
-            if len(target.size()) > 1:
-                target = target.argmax(dim=1, keepdim=True)
-            correct += pred.eq(target.view_as(pred)).sum().item()
+
```
Comment (tremblerz) [inline src/utils/model_utils.py:79]: Can you please make this function modular? It is going to get worse if we add a few more tasks with totally different evaluation. I recommend creating two functions - one for pascal and one for classification (that should be called when else is executed)

File: src/utils/model_utils.py [hunk 2]
```diff
@@ -192,7 +258,45 @@ def test(self, model, dloader, loss_fn, device,
         model.eval()
         test_loss = 0
         correct = 0
-        print("Testing")
+        # TODO: test loader for pascal dataset and object detection
```
Comment (tremblerz) [inline src/utils/model_utils.py:261]: Same modular thing here

File: src/configs/algo_config.py [hunk 3]
```diff
@@ -18,13 +18,13 @@
 
 traditional_fl = {
     "algo": "fedavg",
-    "exp_id": 10,
-    "exp_type": "iid_clients_federated",
+    "exp_id": "no_mixed_precision_more_users",
```
Comment (tremblerz) [inline src/configs/algo_config.py:21]: can you please make a separate config instead of changing our traditional fl

**Review by tremblerz — CHANGES_REQUESTED** [review]

File: src/utils/model_utils.py (hunk 1 above)
Comment (joyce-yuan) [reply src/utils/model_utils.py:79]: :0 done

File: src/configs/algo_config.py (hunk 3 above)
Comment (joyce-yuan) [reply src/configs/algo_config.py]: done

**Review by tremblerz — APPROVED** [review]

**Pull request merged**